### ISSUE TITLE
Add ability to send notifications out when stalking is enabled and triggered

### ISSUE THREAD

**Issue by jomann09** [NagiosEnterprises/nagioscore#605]
This is a follow up to #342 and #604 

Instead of making `stalking_options` create notifications when stalking is logging output changes on checks, there should be a `stalking_notify` option that would, when set to various options, will send a notification when stalking logs the output changes.

This should probably be done in `checks.c` around the host/service event logging section where it checks for stalking currently.

Notification type is up for debate, it may need to be a new notification type, but could also just be a normal style notification that sends out the new output. This is likely the easiest way to do it. May need to update the notification filters to allow it to pass through.

**Comment by Madlohe** [comment]
Doesn't this confuse the difference between state stalking and volatility? Perhaps we could accomplish the same thing by changing is_volatile to volatility_level (or some better name) and letting it have values between 0-3

**Comment by ericloyd** [comment]
State stalking versus volatile has always made less than 100% sense to me.  They're basically the same thing and "volatile" makes it sound like it's "about to explode."  I'd like to see an expansion in the stalking side of things rather than the volatile side of things to make it more useful.

That's a personal preference, not a recommendation.  :-)

**Comment by BanditBBS** [comment]
Jake, it's been a while, any update on the FR?

**Comment by jomann09** [comment]
I added it to the milestones for a new feature version of Core, probably 4.5 if we have one before 5. I am not sure when we will be getting that out though.

**Comment by BanditBBS** [comment]
Any update on this Jake or any other Nagios person can give?

**Comment by ericloyd** [comment]
ca-ricket

**Comment by vladki77** [comment]
Yeah I'd love to see this function working. For those who have no idea how this is useful, I'll give an example. Imagine IPMI monitoring, that combines fans, temperature, voltage, etc. And one of the fans goes off. You have redundancy, so the server is still running, and it takes some time to get a replacement. So you'd like to ACK the service to get rid of repeated notifications, but still you want to be alerted if something changes - e.g. more fans break, or temperatures rise, etc...

A notification when stalking occurs would be nice. But even better would be "Stalking" ACK, similar to "Sticky" ACK. So a stalking event would remove the ACK, and start sending alerts again.

You cannot use "volatile" for this. You do not want notification on every check, quite the contrary. You want notification if the details change, i..e. more things get broken, but the overall state is still the same (WARN/CRIT). 

Also the stalking_option=N is useless, as it reverses the logic. It means "notification causes stalking event". What we need is "stalking event causes notification" or "stalking event removes acknowledgment".
